pkg/kp/dsstore/consul_store.go

```diff
@@ -214,6 +214,84 @@ func (s *consulStore) MutateDS(
 	return ds, nil
 }
 
+// Watch watches dsTree for changes and returns a blocking channel
+// where the client can read a WatchedDaemonSets object which contain
+// changed daemon sets
+func (s *consulStore) Watch(quitCh <-chan struct{}) <-chan WatchedDaemonSets {
+	inCh := make(chan *consulutil.WatchedChanges)
+	outCh := make(chan WatchedDaemonSets)
+	errCh := make(chan error, 1)
+
+	// Watch for changes in the dsTree and deletedDSTree
+	go consulutil.WatchDiff(dsTree, s.kv, inCh, quitCh, errCh)
+
+	go func() {
+		var kvps *consulutil.WatchedChanges
+		for {
+			// Populate kvps by reading inCh and checking for errors
+			select {
+			case <-quitCh:
+				return
+			case err := <-errCh:
+				s.logger.WithError(err).Errorf("WatchDiff returned error, recovered.")
+			case kvps = <-inCh:
+				if kvps == nil {
```

Comment:
is this a thing that can happen? i guess it doesn't hurt to check for it but maybe log it since it would be very odd